### DISCUSSION TITLE
_G API: Community Documentation

### DISCUSSION THREAD

**Discussion by Dimenpsyonal** [Epix-Incorporated/Adonis#603] [General]
# First of all, what is the _G API? 
The _G API is short for Global, which means it can be executed in any script (assuming you have the right settings enabled).

There currently is no (good) documentation on the API, so I decided to make some.

### Settings
To start, you need to have the appropriate settings enabled.
`settings.G_API = true` This needs to be set to true so that the _G.Adonis API can be used at all.
`settings.G_Access = false` This is optional for security, can be left false if you want. This controls wether scripts need an access key to use _G.Adonis.
`settings.G_Access_Key = "Example_Key"` This can be used to set an access key if the above setting is set to true. Example_Key will not work though.
`settings.G_Access_Perms = "Read"` This controls what permissions other scripts have over the _G API, read or write.

#### With _G.Access disabled:
`_G.Adonis.Function(Input, Player(s))`
For example,
`_G.Adonis.Hint("Message", game.Players:GetPlayers())` would create a hint for all players with the word "Message".

#### With _G.Access enabled:
`Functions = _G.Adonis.Access("API Key", "Functions")` - Replace Functions with whatever functions you wish to access.
For example,
`local Admin = _G.Adonis.Access("TEST_KEY_SAMPLE", "Admin")`
`Admin.AddAdmin(game:GetService("Players")["player_name"], 900)` would give Creator admin to the player called "the_name".

## Functions
These are all the functions you can use with _G.Adonis. Most of them are self-explanatory.
IsAdmin
IsBanned
IsMuted
CheckDonor
GetLevel
SetLighting 
SetPlayerLighting
NewParticle
RemoveParticle
NewLocal
MakeLocal
MoveLocal
RemoveLocal
Hint
Message 
RunCommandAsNonAdmin

**Comment by Expertcoderz** [comment]
Gonna formalize _G API docs on the wiki soon.

**Reply by Dimenpsyonal** [reply]
go on. do it

**Reply by AEW745** [reply]
Why is the _G not working for Adonis? I tried it but says attempt to index with nil.

**Reply by annoynumus9** [reply]
@AEW745 i had the same problem but how i solved it was i added the following code to the start of my script so that it waits until _G.Adonis is not nil
```lua
while _G.Adonis==nil do
	wait(0.1)
end
```

**Reply by Dimenpsyonal** [reply]
Interesting

**Reply by jakeb23151** [reply]
> @AEW745 i had the same problem but how i solved it was i added the following code to the start of my script so that it waits until _G.Adonis is not nil
> 
> ```lua
> while _G.Adonis==nil do
> 	wait(0.1)
> end
> ```

this checks if the _g.Adonis is nil "==" it should be "~= nil" to check if its not nil

**Reply by 0x-Offset** [reply]
> > @AEW745 i had the same problem but how i solved it was i added the following code to the start of my script so that it waits until _G.Adonis is not nil
> > ```lua
> > while _G.Adonis==nil do
> > 	wait(0.1)
> > end
> > ```
> 
> this checks if the _g.Adonis is nil "==" it should be "~= nil" to check if its not nil

wrong, they are checking if it IS nil and if it is nil waiting and then checking again, checking if it isnt nil would do the opposite of what they are trying to do.

**Comment by Dimenpsyonal** [comment]
> Gonna formalize _G API docs on the wiki soon.

Yeah right. As if Expertcoderz would do something helpful!

**Comment by LuaScripted** [comment]
if only

**Comment by ghost** [comment]
@StayBlue might interest you

**Comment by JustBeyYT** [comment]
Very good documentation, keep up the good work!

**Comment by PlawanlnwZ** [comment]
Can we get the admin’s rank?
